packages/@lwc/integration-karma/test/custom-elements/index.spec.js

```diff
@@ -858,6 +858,40 @@ if (SUPPORTS_CUSTOM_ELEMENTS) {
                 expect(lwcElm.getAttribute('foo')).toBeNull();
             });
 
+            it('attributeChangedCallback fires on upgrade', () => {
+                // The LWC component observes nothing, but the vanilla component observes foo
+                const tagName = 'x-attr-change-on-upgrade';
+                const lwcElm = createElement(tagName, {
+                    is: ObserveNothing,
```

Comment:
Should you instead use `ObserveNothingThrow` so that the test fails if the lwc element's `attributeChangedCallback` is invoked?

Reply:
Sure, why not. :)